auslib/web/admin/swagger/api.yaml

```diff
@@ -133,6 +379,492 @@ paths:
+        description: Deletes the named Release
+      parameters:
+        - $ref: '#/parameters/releaseParam'
+        - $ref: '#/parameters/csrfTokenParam'
+        - $ref: '#/parameters/dataVersionParam'
+      responses:
+        '200':
+          description: Successfully deleted the release
+        '403':
+          $ref: '#/responses/unauthorizedUser'
+        '404':
+          $ref: '#/responses/resourceNotFound'
+
+  /releases/{release}/read_only:
+    get:
+      summary: Returns whether or not the named Release is marked as read_only
+      description: >
+        Returns whether or not the named Release is marked as read_only. [Docs](http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#releases-release-read-only)
+      tags:
+        - Releases
+      operationId: auslib.web.admin.views.mapper.release_read_only_get
+      consumes: []
+      produces:
+        - application/json
+      externalDocs:
+        url: "http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#releases-release-read-only"
+        description: Returns whether or not the named Release is marked as read_only
+      parameters:
+        - $ref: '#/parameters/releaseParam'
+      responses:
+        '200':
+          description: Returns release object data
+          schema:
+            type: object
+            required:
+              - read_only
+            properties:
+              read_only:
+                description: boolean flag set to True is release is read only else set as False
+                type: boolean
+                example: true
+        '404':
+          $ref: '#/responses/resourceNotFound'
+
+    put:
+      summary: Sets or unsets the read_only flag of the named Release.
+      description: >
+        Sets or unsets the read_only flag of the named Release. Requires release name, data_version and read_only values. [Docs](http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#id15)
+      tags:
+        - Releases
+      operationId: auslib.web.admin.views.mapper.release_read_only_put
+      consumes:
+        - application/json
+      produces:
+        - text/html
+        - application/json
+      externalDocs:
+        url: "http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#id15"
+        description: Sets or unsets the read_only flag of the named Release.
+      parameters:
+        - $ref: '#/parameters/releaseParam'
+        - name: release_read_only_body
+          in: body
+          description: Read Only Release modify Request paramaters
+          required: true
+          schema:
+            allOf:
+              - $ref: '#/definitions/CSRFModel'
+              - $ref: '#/definitions/DataVersionModel'
+              - type: object
+                description: read_only boolean flag
+                required:
+                  - read_only
+                properties:
+                  read_only:
+                    type: ["boolean", "string"]
```

Comment:
the input request json always contains the read_only key when submitted from the UI due to this [statement](https://github.com/aksareen/balrog/blob/f6e7c578b348c5a0d01f8b9f561603a8b3e8f9c1/ui/app/js/controllers/release_read_only_controller.js#L10), hence the blank string '' is treated as type: string instead of type: null.